app/models/maestrano/connector/rails/concerns/connec_helper.rb

```diff
@@ -0,0 +1,96 @@
+module Maestrano::Connector::Rails::Concerns::ConnecHelper
+  extend ActiveSupport::Concern
+
+  module ClassMethods
+    
+    # Replace the ids arrays by the external id
+    # If a reference has no id for this oauth_provider and oauth_uid but has one for connec returns nil
+    def unfold_references(connec_entity, references, organization)
+      connec_entity = connec_entity.with_indifferent_access
```

Comment:
I would personnaly introduce a new variable name, like result, or unfold_connec_entities, to make it clear that there is not side effect on the connec_entity entry variable (personnal taste, I love obviously "pure method")